plutus-ledger-api/src/PlutusLedgerApi/V3/Contexts.hs

```diff
@@ -0,0 +1,577 @@
+  deriving (Pretty) via (PrettyShow HotCommitteeHash)
+  deriving newtype
+    ( PlutusTx.Eq
+    , PlutusTx.ToData
+    , PlutusTx.FromData
+    , PlutusTx.UnsafeFromData
+    )
+
+newtype DRepHash = DRepHash V2.Credential
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+  deriving (Pretty) via (PrettyShow DRepHash)
+  deriving newtype
+    ( PlutusTx.Eq
+    , PlutusTx.ToData
+    , PlutusTx.FromData
+    , PlutusTx.UnsafeFromData
+    )
+
+data DRep
+  = DRepCredential DRepHash
+  | DRepAlwaysAbstain
+  | DRepAlwaysNoConfidence
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+  deriving (Pretty) via (PrettyShow DRep)
+
+instance PlutusTx.Eq DRep where
+  {-# INLINEABLE (==) #-}
+  DRepCredential a == DRepCredential a'            = a PlutusTx.== a'
+  DRepAlwaysAbstain == DRepAlwaysAbstain           = Haskell.True
+  DRepAlwaysNoConfidence == DRepAlwaysNoConfidence = Haskell.True
+  _ == _                                           = Haskell.False
+
+data Delegatee
+  = DelegStake V2.PubKeyHash
+  | DelegVote DRep
+  | DelegStakeVote V2.PubKeyHash DRep
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+  deriving (Pretty) via (PrettyShow Delegatee)
+
+instance PlutusTx.Eq Delegatee where
+  {-# INLINEABLE (==) #-}
+  DelegStake a == DelegStake a' = a PlutusTx.== a'
+  DelegVote a == DelegVote a' = a PlutusTx.== a'
+  DelegStakeVote a b == DelegStakeVote a' b' =
+    a PlutusTx.== a' PlutusTx.&& b PlutusTx.== b'
+  _ == _ = Haskell.False
+
+{- | There is probably no need for Plutus Scripts to even know about all this data in Anchors,
+but it is here for completeness. So we might wanna omit them, especially since urls are a
+bit tricky, since we can't guarantee their validity, not even that they are utf8-encoded.
+-}
+data Anchor = Anchor
+  { anchorUrl      :: PlutusTx.BuiltinString -- Arbitrary Url
+  , anchorDataHash :: PlutusTx.BuiltinByteString -- Blake2b_256 of some off-chain data
+  }
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+
+instance Pretty Anchor where
+  pretty Anchor{..} =
+    vsep
+      [ "anchorUrl:" <+> pretty anchorUrl
+      , "anchorDataHash:" <+> pretty anchorDataHash
+      ]
+
+instance PlutusTx.Eq Anchor where
+  {-# INLINEABLE (==) #-}
+  Anchor a b == Anchor a' b' = a PlutusTx.== a' PlutusTx.&& b PlutusTx.== b'
+
+{- | Note that it no longer is called DCert, since certificates are no longer all about
+delegation, they are also about governance.
+
+Notable changes:
+
+* Pool certificates are unchanged
+* Addition of deposits and refunds. Those values are validated according to respectful
+  PParams. Certificates that where present before get their deposit/refund as
+  optional. This is done for graceful tarnsition, since next era after Conway will require
+  deposit/refund amount
+* New DRep entity
+* New Constitutional Committee entity
+-}
+data TxCert
+  = -- | Register staking credential with an optional deposit amount
+    TxCertRegStaking V2.StakingCredential (Haskell.Maybe V2.Value)
```

Comment:
@lehins @WhatisRT There's no `StakingHash`; I'm assuming you meant `StakingCredential`.

Reply:
Yeah. Same is true for the `DRepHash`, it's a credential as well.

Reply:
`StakingCredential` is different from `Credential`. `StakingCredential` also contains pointers and and can be null:
https://github.com/input-output-hk/plutus/blob/a30229b3a07f47472460e5fe8c6135442d5c56a7/plutus-ledger-api/src/PlutusLedgerApi/V1/Credential.hs#L26-L38

Where Credential can be either KeyHash or ScriptHash. This one needs to be Credential
```suggestion
    TxCertRegStaking V2.Credential (Haskell.Maybe V2.Value)
```